numba/cuda/tests/cudapy/test_dispatcher.py

```diff
@@ -93,6 +104,116 @@ def f(x, y):
         self.assertEqual(len(f.specializations), 2)
         self.assertIs(f_f32c_f32c, f_f32c_f32c_2)
 
+    # The following tests are based on those in numba.tests.test_dispatcher
```

Comment:
Note: I've checked against the originals and these new translated tests are a faithful representation of those tests in `numba.tests.test_dispatcher`.